AppDB/appscale/datastore/zkappscale/entity_lock.py

```diff
@@ -29,23 +29,23 @@ def zk_group_path(key):
   Returns:
     A string containing the location of a ZooKeeper path.
   """
-  project = key.app()
+  project = key.app().decode('utf-8')
   if key.name_space():
-    namespace = key.name_space()
+    namespace = key.name_space().decode('utf-8')
   else:
-    namespace = ':default'
+    namespace = u':default'
 
   first_element = key.path().element(0)
-  kind = first_element.type()
+  kind = first_element.type().decode('utf-8')
 
   # Differentiate between types of identifiers.
   if first_element.has_id():
-    suffix = ':{}'.format(first_element.id())
+    group = u'{}:{}'.format(kind, first_element.id())
```

Comment:
IDs are automatically allocated integers. Names are user-defined strings. An entity key can have either one or the other.